### PR TITLE
Fix: dtdl 'extends' is a list

### PR DESCRIPTION
**Pull request by csm-pjm**

### CODE REVIEW

File: src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt [hunk 1]
```diff
@@ -52,13 +41,14 @@ class JsonUtil {
          * @return a Pair(Boolean,String) containing minimal extension information
          */
         @JvmStatic
-        fun isExtension(jsonModel: JsonObject):Pair<Boolean,String> {
-            return if( jsonModel.containsKey(DTDL_EXTENDS_KEY) )
-                Pair(true,jsonModel[DTDL_EXTENDS_KEY].toString())
+        fun isExtension(jsonModel: JsonObject):Pair<Boolean,List<String>> {
+            val result = jsonModel.array<String>(DTDL_EXTENDS_KEY)
```
Comment (rm3l) [inline src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt:45]: This line will throw an error if `extends` is a String, which remains a valid form.
Per [this page](https://docs.microsoft.com/en-us/azure/digital-twins/concepts-models#model-inheritance) and https://github.com/Azure/opendigitaltwins-dtdl/issues/104#issuecomment-738207892, `extends` can be either a String or an Array.

These forms reportedly are all valid:

- `"extends": "dtmi:com:example:Base;1"`
- `"extends": [ "dtmi:com:example:Base;1" ]`
- `"extends": [ "dtmi:com:example:Base;1", "dtmi:com:example:AnotherInterface;1" ]`

So we need to support these cases then.

Please consider adding test cases in `JsonUtilsTest`.

File: src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt [hunk 2]
```diff
@@ -52,13 +41,14 @@ class JsonUtil {
...
+            return if(result != null)
+                Pair(true, result)
             else
-                Pair(false,"")
+                Pair(false, listOf<String>())
```
Comment (rm3l) [inline src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt:49]: No need to repeat `String` here, wich will be inferred. So just `listOf()` should be fine, I think.

File: src/test/resources/utils/dt-with-extends.json [hunk 3]
```diff
@@ -2,7 +2,10 @@
   "@context": "dtmi:dtdl:context;2",
   "@id": "dtmi:com:cosmotech:supply:TransportOperation;1",
   "@type": "Interface",
-  "extends": "dtmi:com:cosmotech:supply:Operation;1",
```
Comment (rm3l) [inline src/test/resources/utils/dt-with-extends.json:5]: This remains a valid form for `extends`. So I suggest keeping this as is, so it is tested, but add your changes in a separate file and test case.

**Review by rm3l — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt (hunk 1 above)
Comment (csm-pjm) [reply src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt]: done

File: src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt (hunk 2 above)
Comment (csm-pjm) [reply src/main/kotlin/com/cosmotech/connector/adt/utils/JsonUtil.kt]: done

File: src/test/resources/utils/dt-with-extends.json (hunk 3 above)
Comment (csm-pjm) [reply src/test/resources/utils/dt-with-extends.json]: done

**Comment by rm3l** [comment]
There are few tests not passing : https://github.com/Cosmo-Tech/azure-digital-twins-simulator-connector/pull/27/checks

**Review by rm3l — COMMENTED** [review]
There are few tests not passing : https://github.com/Cosmo-Tech/azure-digital-twins-simulator-connector/pull/27/checks

**Review by rm3l — CHANGES_REQUESTED** [review]

File: src/test/kotlin/com/cosmotech/connector/utils/JsonUtilTest.kt [hunk 4]
```diff
@@ -28,42 +28,32 @@ class JsonUtilTest: AbstractUnitTest() {
...
             StringReader(
-                getResourceFile("utils/dt-with-no-extends.json")
+                getResourceFile("utils/dt-with-list-extends.json")
                     .toFile()
                     .readText(Charsets.UTF_8)
             )
         )
 
-    @Test
-    fun test_readPropertiesName() {
```
[the remark this answers is not in the document]
Comment (csm-pjm) [reply src/test/kotlin/com/cosmotech/connector/utils/JsonUtilTest.kt:50]: Parce que j'ai nettoyé readPropertiesName qui n'était pas utilisée et semblait un doublon par rapport à readPropertiesNameAndType

**Review by rm3l — APPROVED** [review]

**Review by rm3l — APPROVED** [review]

**Pull request merged**